src/service/cid-api.js

```diff
@@ -123,6 +129,7 @@ export class GoogleCidApi {
   fetchCid_(url, scope, token) {
     const payload = dict({
       'originScope': scope,
+      'canonicalOrigin': this.canonicalOrigin_,
```

Comment:
@Gregable theoretically canonicalOrigin can be `null` if there is misconfiguration (might only happen when the page is invalid AMP). what does CID API server do in that case?

Reply:
It should ignore the field and use the Host header instead.